arangod/Cluster/ServerState.cpp

```diff
@@ -695,63 +702,105 @@ bool ServerState::checkEngineEquality(AgencyComm& comm) {
 /// @brief check equality of naming conventions settings with other registered
 /// servers
 bool ServerState::checkNamingConventionsEquality(AgencyComm& comm) {
-  // our own setting
-  bool const extendedNamesForDatabases =
-      _server.getFeature<DatabaseFeature>().extendedNamesForDatabases();
-
   AgencyCommResult result = comm.getValues(::currentServersRegisteredPref);
   if (result.successful()) {  // no error if we cannot reach agency directly
-
     auto slicePath =
         AgencyCommHelper::slicePath(::currentServersRegisteredPref);
     VPackSlice servers = result.slice()[0].get(slicePath);
     if (!servers.isObject()) {
       return true;  // do not do anything harsh here
     }
 
-    for (VPackObjectIterator::ObjectPair pair : VPackObjectIterator(servers)) {
-      if (!pair.value.isObject()) {
-        continue;
-      }
-      VPackSlice setting = pair.value.get(::extendedNamesDatabasesKey);
-      if (setting.isBool() && setting.getBool() != extendedNamesForDatabases) {
-        // different setting detected. bail out!
-        LOG_TOPIC("75972", FATAL, arangodb::Logger::STARTUP)
-            << "The usage of different settings for database object naming "
-            << "conventions (i.e. `--database.extended-names-databases` "
-               "settings) "
-            << "in the cluster is unsupported and may cause follow-up issues. "
-            << "Please unify the settings for the startup option "
-               "`--database.extended-names-databases` "
-            << "on all coordinators and DB servers in this cluster.";
-
-        std::string msg;
-        for (VPackObjectIterator::ObjectPair p : VPackObjectIterator(servers)) {
-          if (!p.value.isObject()) {
-            continue;
+    auto checkSetting = [](velocypack::Slice servers,
+                           std::string_view optionName, std::string_view key,
+                           bool localValue) -> bool {
+      for (auto pair : VPackObjectIterator(servers)) {
+        if (!pair.value.isObject()) {
```

Comment:
I don't think so. This is startup code, and refusal of startup is very bad because it will cause endless startup loops e.g. in ArangoGraph. We should always try to handle issue gracefully, IMHO.